### PR TITLE
feat(april-fools): add `I'm Feeling Lucky`

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Add `I'm feeling lucky`

image

Add the [Lucky Token list](https://tokenlists.org/token-list?url=https://dpaste.com/3SCBYM7GS.txt) based on coingecko top ~100~ 96 meme coins, enabled by default on mainnet
image

# To Test

1. With the feature flag `isAprilFoolsEnabled` on, load the page
* I'm feeling lucky should be displayed in the button
* There should be no buy token element in the UI
2. Click on `I'm feeling lucky`
* A random token should be selected as BUY
* The new slot machine sound should play
* Flow should continue as usual (SWAP if there is sell funds, and so on)

3. With the feature flag `isAprilFoolsEnabled` off, load the page
* There should NOT be a `I'm feeling lucky` button
* App should work as usual

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @alfetopito , still testing, but already have some questions: 
1. Should we keep the form without a But token when not connected?
image

2. TWAP/Limit orders forms also don't have To token
image
image

3. When on a reduced screen, see how the trade mode looks like when there is no 'To' field
image

4. Is this OK to return to the 'default' state (with no To field) when press on the Home button?

**Comment by alfetopito** [comment]
@elena-zh 
Fixed 1,2,5
3 is out of the scope
4 is ok IMO
6 is being worked on by Michel

File: apps/cowswap-frontend/src/common/pure/CurrencyInputPanel/CurrencyInputPanel.tsx [hunk 1]
```diff
@@ -151,7 +153,7 @@ export function CurrencyInputPanel(props: CurrencyInputPanelProps) {
         {topLabel && <styledEl.CurrencyTopLabel>{topLabel}</styledEl.CurrencyTopLabel>}
 
         <styledEl.CurrencyInputBox>
-          <div>
+          <div id={wasImFeelingLuckyClicked ? 'fuck' : 'yuck'}>
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/common/pure/CurrencyInputPanel/CurrencyInputPanel.tsx:156]: @fairlighteth this is the flag to use.
Please clear this id 😬

File: libs/tokens/src/const/tokensList.json [hunk 2]
```diff
@@ -10,6 +10,11 @@
       "enabledByDefault": true,
       "source": "https://files.cow.fi/tokens/CoinGecko.json"
     },
+    {
+      "priority": 13,
+      "enabledByDefault": true,
+      "source": "https://dpaste.com/3SCBYM7GS.txt"
```
Comment (shoom3301) [inline libs/tokens/src/const/tokensList.json:16]: Should we add similar list to Gnosis chain? Or all "lucky" tokens already included in the default lists?

**Review by shoom3301 — APPROVED** [review]

**Review by elena-zh — APPROVED** [review]
Changes LGTM!
Still there is an issue with waving TO token on SWAP/Limit/TWAP forms (reported in https://github.com/cowprotocol/cowswap/pull/4099#issuecomment-2023522349), but we can survive 1 day with this. 

And fix [this ](https://cowservices.slack.com/archives/C0361CDG8GP/p1711627197377209?thread_ts=1711622447.311619&cid=C0361CDG8GP) issue

**Pull request merged**